### PR TITLE
Empty group should not be clickable and not be visited

### LINKED ISSUE #9427: [Med] Empty group should not be clickable and be visited
For retesting:
Configure WCEM_MULTILEVEL.
Click on **empty** or **should be invisible** under **java part** groups.

**Expectation**:
A group without any attributes should not be visitable or clickable. It means such a group is not reachable neither via **previous** or **next buttons navigation** nor via **group menu navigation**.
The styling of the unreachable group is **_requested_**. The current styling is the unreachable group gets a trash icon and by hovering over group title no cursor is available.

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts [hunk 1]
```diff
@@ -371,4 +376,16 @@ describe('ConfigurationGroupMenuComponent', () => {
     expect(configuratorGroupsService.isGroupVisited).toHaveBeenCalled();
     expect(configuratorGroupsService.getGroupStatus).toHaveBeenCalled();
   });
+
+  it('should verify whether the group is disable', () => {
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts]: Small typo: should verify whether the group is disabled

File: feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts [hunk 2]
```diff
@@ -371,4 +376,16 @@ describe('ConfigurationGroupMenuComponent', () => {
...
+    productConfigurationObservable = of(mockProductConfiguration);
+    routerStateObservable = of(mockRouterState);
+    initialize();
+    component
+      .isGroupDisabled(mockProductConfiguration.owner, GROUP_ID_8)
+      .pipe(take(1))
+      .subscribe((isGroupDisable) => {
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts]: isGroupDisabled

File: feature-libs/product/configurators/common/core/model/configurator.model.ts [hunk 3]
```diff
@@ -173,5 +173,6 @@ export namespace Configurator {
   export enum GroupStatus {
     COMPLETE = 'COMPLETE',
     ERROR = 'ERROR',
+    DISABLE = 'DISABLE',
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/model/configurator.model.ts]: DISABLED = DISABLED

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts [hunk 4]
```diff
@@ -584,4 +587,88 @@ describe('ConfiguratorEffect', () => {
       expect(configEffects.getGroupWithAttributes(groups)).toBeUndefined();
     });
   });
+
+  it('should collect disabled group IDs', () => {
+    const disableGroupIds = [];
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts]: disabledGroupIds

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts [hunk 5]
```diff
@@ -58,6 +73,21 @@ export class ConfiguratorBasicEffects {
     })
   );
 
+  collectDisabledGroupIds(
+    disableGroupIds: string[],
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts]: disabledGroupIds

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts [hunk 6]
```diff
@@ -58,6 +73,21 @@ export class ConfiguratorBasicEffects {
     })
   );
 
+  collectDisabledGroupIds(
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts]: Instance method needs to be after all instance fields (effects)!. I get a warning in VS code for that

Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts]: And why do we need this at all? Why can't we just check for the 'configurable' attribute in the groups?

File: feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.html [hunk 7]
```diff
@@ -23,10 +23,16 @@
             }}"
             [class.cx-menu-conflict]="isConflictGroupType(group.groupType)"
             [class.WARNING]="isConflictGroupType(group.groupType)"
+            [class.DISABLE]="
+              isGroupDisabled(configuration.owner, group.id) | async
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.html]: Why do we need this? We are looping the groups and could just query the configurable attribute, or am I missing something?

**Review by ChristophHi — CHANGES_REQUESTED** [review]
I do have doubts about the overall concept, is it really needed to introduce new actions, reducers etc? See my comments..

File: feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts]: Done.

File: feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/components/group-menu/configurator-group-menu.component.spec.ts]: Done.

File: feature-libs/product/configurators/common/core/model/configurator.model.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/model/configurator.model.ts]: Done

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts]: Done

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts]: Done

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.ts]: Done

File: feature-libs/product/configurators/common/core/facade/configurator-group-status.service.ts [hunk 8]
```diff
@@ -222,25 +221,25 @@ export class ConfiguratorGroupStatusService {
...
       return;
     }
 
-    //Currently only check for completness, no validation of input types
+    //Currently only check for completeness, no validation of input types
     if (this.checkIsGroupComplete(group)) {
       completedGroupIds.push(group.id);
     } else {
-      incompleteOrErrorGroupdIds.push(group.id);
+      incompleteOrErrorGroupIds.push(group.id);
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/facade/configurator-group-status.service.ts:235]: Good that you corrected those typos...

File: feature-libs/product/configurators/common/core/state/actions/configurator.action.ts [hunk 9]
```diff
@@ -49,6 +49,7 @@ export const SET_MENU_PARENT_GROUP =
 export const SET_GROUPS_VISITED = '[Configurator] Set groups to visited';
 export const SET_GROUPS_COMPLETED = '[Configurator] Set groups complete status';
 export const SET_GROUPS_ERROR = '[Configurator] Set groups error status';
+export const SET_GROUPS_DISABLE = '[Configurator] Set groups disabled status';
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/actions/configurator.action.ts]: This we don't need anymore!

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts [hunk 10]
```diff
@@ -35,12 +35,14 @@ const owner: GenericConfigurator.Owner = {
 
 const group: Configurator.Group = {
   id: groupId,
+  configurable: true,
```
Comment (ChristophHi) [inline feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts]: We don't need these changes, right?

**Review by ChristophHi — CHANGES_REQUESTED** [review]
There is one obsolete action ID left...

File: feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts (hunk 10 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/state/effects/configurator-basic.effect.spec.ts]: Right!

File: feature-libs/product/configurators/common/core/state/actions/configurator.action.ts (hunk 9 above)
Comment (Larisa-Staroverova) [reply feature-libs/product/configurators/common/core/state/actions/configurator.action.ts]: Done

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**